src/utils.ts

```diff
@@ -93,16 +79,23 @@ export async function getData({
   });
 
   if (!response.ok) {
-
-    const errorBody = await response.json();
-    const error = {
-      status: response.status,
-      message: errorBody.message || 'API request failed',
-    };
-
-    throw error;
+    throw (await makeError(response));
   }
 
   const result = await response.json();
   return result;
+}
+
+async function makeError(response: Response) {
+  let errorBody: any;
+  try {
+    errorBody = await response.json();
+  } catch (_) {
+    errorBody = null;
+  }
+
+  return {
+    status: response.status,
```

Comment:
in case if response json fail, `response.status` will be undefined, so let's set code 500 as fallback here

```suggestion
    status: response.status || 500,
```